core/src/main/java/org/apache/calcite/sql/SqlIfStmt.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql;
+
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.util.ImmutableNullableList;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Parse tree for a {@code SqlIfStmt}.
+ */
+public class SqlIfStmt extends SqlCall {
+  public static final SqlSpecialOperator OPERATOR =
+      new SqlSpecialOperator("IF", SqlKind.CONDITIONAL_STATEMENT);
+
+  public final SqlNodeList conditionalStmtListPairs;
+  public final SqlNodeList elseStmtList;
+
+  /**
+   * Creates a {@code SqlTablePartitionByColumn}.
+   * @param pos                       Parser position, must not be null.
+   * @param conditionalStmtListPairs  List of conditional expression pairs
+   *                                  with StatementList.
+   * @param elseStmtList              List of statements in the else clause.
+   */
+  public SqlIfStmt(final SqlParserPos pos,
+      final SqlNodeList conditionalStmtListPairs,
+      final SqlNodeList elseStmtList) {
+    super(pos);
+    Objects.requireNonNull(conditionalStmtListPairs);
+    this.conditionalStmtListPairs = conditionalStmtListPairs;
+    this.elseStmtList = elseStmtList;
+  }
+
+  @Override public SqlOperator getOperator() {
+    return OPERATOR;
+  }
+
+  @Override public List<SqlNode> getOperandList() {
+    return ImmutableNullableList.of(conditionalStmtListPairs, elseStmtList);
+  }
+
+  @Override public void unparse(final SqlWriter writer, final int leftPrec,
+      final int rightPrec) {
+    for (int i = 0; i < conditionalStmtListPairs.size(); i++) {
+      if (i != 0) {
+        writer.keyword("ELSE IF");
+      } else {
+        writer.keyword("IF");
+      }
```

Comment:
You could potentially extract the "IF" keyword since it is present in both cases (so you would not need the else clause). Not a big deal, just a thought.

Reply:
Thanks for the information on how to use Objects.requireNonNull(). I've checked Effective Java, and figured out I should start using it on all the Java classes methods' parameters that have non-null restriction.

Reply:
For the "IF" keyword printing, I believe the existing logic looks more readable so I'll keep it. Unless some other reviewers have strong opinion on this.

Reply:
Leaving as is SGTM

Reply:
Fine either way, totally up to you.